rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -140,6 +140,14 @@ void SequentialWriter::reset()
     metadata_io_->write_metadata(base_folder_, metadata_);
   }
 
+  // if cache data exists, it must flush the data into the storage
+  if (!cache_.empty()) {
+    storage_->write(cache_);
+    // reset cache
```

Comment:
Since this code is used as is in multiple places now, can be put this inside a helper to avoid duplication? maybe a `reset_cache` method?

Reply:
sounds good, will do that.